qiita_db/data.py

```diff
@@ -1034,13 +1039,16 @@ def update_vamps_status(self, status):
         ValueError
             If the status is not known.
         """
-        if status not in ('not submitted', 'submitting', 'success', 'failed'):
-            raise ValueError("Unknown status: %s" % status)
+        with TRN:
+            if status not in ('not submitted', 'submitting', 'success',
```

Comment:
You're correct, there is not DB access in that line, so it can be outside of the transaction started here
Good catch!